tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -216,6 +216,14 @@ bpf_code_generator::parse()
     ELFIO::const_symbol_section_accessor symbols{reader, reader.sections[".symtab"]};
 
     if (map_section) {
+        size_t data_size = map_section->get_size();
+
+        if (data_size % sizeof(ebpf_map_definition_in_file_t) != 0) {
```

Comment:
In [verifier PR 295](https://github.com/vbpf/ebpf-verifier/pull/295) I put a more intelligent algorithm for determining the map definition size in the file in https://github.com/vbpf/ebpf-verifier/blob/master/src/asm_files.cpp#L79 which is the algorithm already used by libbpf.

Consider using that algorithm, which supports older (smaller) structures by 0-padding the rest of the fields.